tests/unit/util/pack_traversal.cpp

```diff
@@ -0,0 +1,621 @@
+//  Copyright (c) 2017 Denis Blank
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <algorithm>
```

Comment:
Usually I try to stick to the Google C++
style guide (https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) which requests to include standard library headers first for the reason:
>With the preferred ordering, if dir2/foo2.h omits any necessary includes, the build of dir/foo.cc or dir/foo_test.cc will break. Thus, this rule ensures that build breaks show up first for the people working on these files, not for innocent people in other packages.

But I will reorder the includes to keep the conformance to the existing code.